galley/pkg/config/source/kube/apiserver/source.go

```diff
@@ -193,6 +193,10 @@ func (s *Source) startWatchers() {
 	scope.Source.Info("Creating watchers for Kubernetes CRDs")
 	s.watchers = make(map[collection.Name]*watcher)
 	for i, r := range resources {
+		if s.provider == nil {
```

Comment:
Need a Mutex to guard against `s.provider` going nil between this line and the use of it to create `a` on line 200?

Reply:
This function is already only called while the `mu` lock is held, so it cannot be set to nil between these lines.